File: BE/src/main/java/com/hr_management/hr_management/entity/LeaveBalance.java
package com.hr_management.hr_management.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Entity
@Table(name = "LeaveBalance")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class LeaveBalance {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "leave_balance_id")
    private Integer leaveBalanceId;

    @Column(name = "year", nullable = false)
    private Integer year;

    @Column(name = "used_leave")
    private Integer usedLeave = 0;

    @Column(name = "remaining_leave", nullable = false)
    private Integer remainingLeave;

    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "employee_id", nullable = false)
    private Employee employee;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "leave_type_id", nullable = false)
    private LeaveType leaveType;
}

File: BE-dotnet/Models/LeaveBalance.cs
using System;

namespace EmpowerHR.Models
{
    public class LeaveBalance
    {
        public int LeaveBalanceId { get; set; }
        public int Year { get; set; }
        public int UsedLeave { get; set; }
        public int RemainingLeave { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime? UpdatedAt { get; set; }
        public int EmployeeId { get; set; }
        public int LeaveTypeId { get; set; }

        // Navigation Properties
        public Employee Employee { get; set; }
        public LeaveType LeaveType { get; set; }
    }
}
